.github/workflows/typos_spellcheck.yml

```diff
@@ -0,0 +1,33 @@
+# Reference: https://github.com/crate-ci/typos/blob/master/docs/github-action.md
+
+name: 🔡 Spell Check
+
+on:
+  push:
+    branches:
+      - main
+      # TODO: (decide) Update this to run on all branches
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  run:
+    name: 🔍 Spell Check with Typos
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Actions Repository
+        uses: actions/checkout@v3
+
+      - name: Use typos config file
+        uses: crate-ci/typos@master
```

Comment:
How much time does this usually take to run? are we running this only on staged files in that PR or the complete repo for every pr?

Reply:
Ohh! I forgot to mention this in What's next? section... adding now.

Definitely;

1. we should be running the check on the changed files only.
2. time for whole repo check ~ 3-5 sec, using pre-built bin may help reduce it further for whole repo check
3. for checking only diff w/ above thing in place no more than 2 sec I would say
4. excluding, time req for commenting the o/p.